### PR TITLE
Add timer, game over screen, several events

### PR DESCRIPTION
**Pull request by Faffie**
Moved constants/settings to seperate Settings class.
Implement timer.
Timer value can be overwritten by host.
Added state for when game is lost.
Several new events to control game.
Moved overlays to assets/Interface.
Refactored states. 
Minor other improvements.

Please review and merge. Merging PR on host is required.

### CODE REVIEW

File: EscapeVR/src/main/java/nl/tudelft/kroket/escape/EscapeVR.java [hunk 1]
```diff
@@ -53,20 +57,12 @@
...
-  private static final Vector3f spawnPosition = new Vector3f(0, 0, 0);
-
   /** Current class, used as tag for logger. */
   private final String className = this.getClass().getSimpleName();
 
   /** Singleton logger instance. */
   private Logger log = Logger.getInstance();
 
-  /** Portnumber of the gamehost. */
-  private static int PORTNUM = 1234;
```
Comment (Faffie) [inline EscapeVR/src/main/java/nl/tudelft/kroket/escape/EscapeVR.java:71]: Constant moved to Settings.java.

File: EscapeVR/src/main/java/nl/tudelft/kroket/screen/HeadUpDisplay.java [hunk 2]
```diff
@@ -33,7 +33,6 @@
 
   // clear text after 8 seconds
```
Comment (Faffie) [inline EscapeVR/src/main/java/nl/tudelft/kroket/screen/HeadUpDisplay.java:34]: Should probably be moved to Settings

**Comment by Mayke93** [comment]
we'll test and solve the merge conflicts before it can be merged

**Comment by irenevanderblij** [comment]
The timer looks good and works! However, I do not really understand some parts. When looking at this code and the code on the host, I believe you want to keep track of the time on the server (which seems like a good design to me), but the update method of the host is not used, and the VR decides for itself when the game is over

**Comment by Faffie** [comment]
Well, yeah. I let the VR client keep track of the time locally and only sync them at the start of the game. I forgot to push a change in the host though. 

These time functions aren't too reliable, and of course things would get out of sync but I figured it wouldn't matter much. We can sync the timers more often if this really is a problem.

The reason I do want to keep track of the timer at both ends is because I want the VR client to be able to run detached from the server, and because it has to update the label every second, it's better to keep track of it locally instead of having the host update the value.

Whatever, if this really turns out to be a big issue we can easily tweak things.

**Comment by Faffie** [comment]
Resolved merge conflicts.

**Comment by irenevanderblij** [comment]
Gamewon and gamelost screens both work now. Please review and merge!

**Comment by Faffie** [comment]
:+1: 

Works nicely.

**Comment by alanvanrossum** [comment]
heel mooi

**Pull request merged**